src/vm/assembly.cpp

```diff
@@ -2722,7 +2722,7 @@ INT32 Assembly::ExecuteMainMethod(PTRARRAYREF *stringArgs)
     //to decide when the process should get torn down.  So, don't call it from
     // AppDomain.ExecuteAssembly()
     if (pMeth) {
-        if (stringArgs == NULL)
+        if (waitForOtherThreads)
```

Comment:
I am sorry for this stupid bug. I'll fix it in a minute. What you said is true except for one case - the call from corhost.cpp@1348, where the args are not null and we want it to wait.